### PR TITLE
[FSDE-9222] Improved Vue documentation theme switcher

### PR DESCRIPTION
**Pull request by janibolkvadze**

### CODE REVIEW

File: src/documentation/components/ThemeSwitcher.vue [hunk 1]
```diff
@@ -80,24 +80,68 @@
...
 export default class ThemeSwitcher extends Vue {
-  theme: Themes = this.$store.state.themes.theme;
   themes = THEMES;
 
-  setTheme(theme: Themes) {
+  setTheme(theme: Themes): void {
+    const brandLogo = document.getElementById('header-logo') as any;
+    const assetsToReplace : AssetsToReplace[] = [{type: 'css', id: 'chi-css'}, {type: 'docsCss', id: 'chi-docs-css'}];
+
+    assetsToReplace.forEach((asset: { type: 'css' | 'docsCss', id: string }) => {
```
Comment (evansito) [inline src/documentation/components/ThemeSwitcher.vue:119]: Could this be replaced with `forEach((asset: AssetsToReplace)` ?

File: src/documentation/constants/constants.ts [hunk 2]
```diff
@@ -1,17 +1,28 @@
+import { CHI_VERSION, DEFAULT_THEME } from './configs';
+
+export const chiAssetsRootUrl = 'https://assets.ctl.io/chi/' + CHI_VERSION + '/';
```
Comment (evansito) [inline src/documentation/constants/constants.ts:3]: any reason there isn't string interpolation here?

File: src/documentation/components/ThemeSwitcher.vue [hunk 3]
```diff
@@ -80,24 +80,68 @@
...
 
 declare const chi: any;
+interface AssetsToReplace {
+  type: 'css' | 'docsCss';
+  id: string;
+}
 
-@Component({})
+@Component({
+  head() {
```
Comment (evansito) [inline src/documentation/components/ThemeSwitcher.vue:93]: I'm getting a "No overload matches this call." error in head().

File: src/documentation/components/ThemeSwitcher.vue [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <template>
```
Comment (evansito) [inline src/documentation/components/ThemeSwitcher.vue:1]: Formatting looks off, was prettier run on this file?

File: src/documentation/components/ThemeSwitcher.vue [hunk 5]
```diff
@@ -80,24 +80,68 @@
...
+      const replacementHref = THEMES[theme][asset.type];
+
+      if (currentAsset && replacementAsset) {
+        replacementAsset.setAttribute('rel', 'stylesheet');
+        replacementAsset.setAttribute('href', replacementHref);
+        if (currentAsset.parentNode) {
+          currentAsset.parentNode
+            .insertBefore(replacementAsset, currentAsset.nextSibling);
+        }
+        replacementAsset.addEventListener('load', function() {
```
Comment (evansito) [inline src/documentation/components/ThemeSwitcher.vue:131]: Maybe we could use arrow function () => ?

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-224/1/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://526b3d06f3d23370940d992017e89c47a5af1c22-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-224-ux-chi.kubeodc-test.corp.intranet/ (internal)

File: src/documentation/components/ThemeSwitcher.vue (hunk 3 above)
Comment (janibolkvadze) [reply src/documentation/components/ThemeSwitcher.vue:93]: Do you mean in IDE?
Standard Vue extensions of visual studio code don't support Nuxtjs methods like this one and mark it as error.

Here is Nuxt.js documentation https://nuxtjs.org/docs/components-glossary/head/

File: src/documentation/components/ThemeSwitcher.vue (hunk 4 above)
Comment (janibolkvadze) [reply src/documentation/components/ThemeSwitcher.vue:1]: The file is formatted using prettier. No issues in linting
Screenshot 2021-11-05 at 15 15 04

File: src/documentation/components/ThemeSwitcher.vue (hunk 1 above)
Comment (janibolkvadze) [reply src/documentation/components/ThemeSwitcher.vue:119]: Yes, good catch

File: src/documentation/components/ThemeSwitcher.vue (hunk 4 above)
Comment (evansito) [reply src/documentation/components/ThemeSwitcher.vue:1]: Ack!

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-224/2/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-224/3/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://45170ba2547d17e62716712f065efe904718450e-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-224-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Review by evansito — APPROVED** [review]

**Review by josecarlosgonzalezv — APPROVED** [review]

**Pull request merged**